main.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"strconv"
+)
+
+func main() {
+	file, err := os.Open("instance.sat")
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+
+	var expression [][]int
+
+	var clauseIndex = 0
+
+	for scanner.Scan() {
+		line := scanner.Text()
+		switch firstChar := line[:1]; firstChar {
+		case "c":
+			// ignore comment lines
+		case "p":
+			// read numVars and numClauses from p(repare?) line
+			tokens := strings.Split(line, " ")
+			totalNumVars, _ := strconv.Atoi(tokens[2])
```

Comment:
We can't just ignore errors.  I'd make a separate `parse` function that takes an input reader, output writer and returns an 'err' if there's an error.